libraries/transformer/src/main/java/androidx/media3/transformer/DefaultDecoderFactory.java

```diff
@@ -135,6 +135,9 @@ public DefaultCodec createForVideoDecoding(
     if (SDK_INT >= 31 && requestSdrToneMapping) {
       mediaFormat.setInteger(
           MediaFormat.KEY_COLOR_TRANSFER_REQUEST, MediaFormat.COLOR_TRANSFER_SDR_VIDEO);
+    } else if (SDK_INT >= 31 && MimeTypes.VIDEO_DOLBY_VISION.equals(format.sampleMimeType)) {
```

Comment:
We generally write the condition in reversed way `format.sampleMimeType.equals(MimeTypes.VIDEO_DOLBY_VISION)`

Reply:
There's lots of occurrences of the ['yoda condition'](https://en.wikipedia.org/wiki/Yoda_conditions) in our codebase, but I agree with @SheenaChhabra it is now discouraged in new code. In this case, because `Format.sampleMimeType` is `@Nullable` you will need to use `Objects.equals(format.sampleMimeType, MimeTypes.VIDEO_DOLBY_VISION)`.

Reply:
Done.